robot-command/src/main/java/org/obolibrary/robot/ReportCommand.java

```diff
@@ -99,26 +101,21 @@ public CommandState execute(CommandState state, String[] args) throws Exception
     state = CommandLineHelper.updateInputOntology(ioHelper, state, line);
     OWLOntology ontology = state.getOntology();
 
+    // Override default report options with command-line options
+    Map<String, String> reportOptions = ReportOperation.getDefaultOptions();
+    for (String option : reportOptions.keySet()) {
+      if (line.hasOption(option)) {
+        reportOptions.put(option, line.getOptionValue(option));
+      }
+    }
+
     // output is optional - no output means the file will not be written anywhere
     String outputPath = CommandLineHelper.getOptionalValue(line, "output");
-    // profile is optional - no profile means the default profile will be used
-    String profilePath = CommandLineHelper.getOptionalValue(line, "profile");
-    // format is optional - default is TSV, but if the prefix of file is 'csv' or 'yaml',
-    // that will be automatically detected
-    String format = CommandLineHelper.getOptionalValue(line, "format");
-    if (format == null && outputPath != null) {
-      format = outputPath.substring(outputPath.lastIndexOf(".") + 1);
-    }
-    // fail-on is optional - if null, will always exit with 0
-    String failOn = CommandLineHelper.getDefaultValue(line, "fail-on", "error");
-    boolean useLabels = CommandLineHelper.getBooleanValue(line, "labels", false);
 
     // Success is based on failOn
     // If any violations are found of the fail-on level, this will be false
-    // If failOn is "none" or if no violations are found, this will be true
-    boolean success =
-        ReportOperation.report(
-            ontology, ioHelper, profilePath, outputPath, format, failOn, useLabels);
+    // If fail-on is "none" or if no violations are found, this will be true
+    boolean success = ReportOperation.report(ontology, ioHelper, outputPath, reportOptions);
     if (!success) {
       logger.error("Report failed!");
       System.exit(1);
```

Comment:
Is there a reason that we exit like this rather than using the new error handling system?

Reply:
I did it this way because I didn't want the exception formatting from the error handling - I think the printed output would start to look cluttered. I can change it if you think it would be better that way.